app/uk/gov/hmrc/exports/controllers/DeclarationController.scala

```diff
@@ -72,13 +72,20 @@ class DeclarationController @Inject()(
       }
     }
 
-  def findAll(status: Option[String], pagination: Page): Action[AnyContent] =
+  def findAll(
+    status: Option[String],
+    pagination: Page,
+    sortBy: Option[String],
+    sortDirection: Option[String]
+  ): Action[AnyContent] =
     authenticator.authorisedAction(parse.default) { implicit request =>
       val search = DeclarationSearch(
         eori = request.eori.value,
         status = status.map(v => Try(DeclarationStatus.withName(v))).filter(_.isSuccess).map(_.get)
       )
-      declarationService.find(search, pagination).map(results => Ok(results))
+      def direction: Option[String] => Int = option => if (option.exists(_.equalsIgnoreCase("asc"))) 1 else -1
+      val sort = sortBy.map(field => DeclarationSort(field, direction(sortDirection))).getOrElse(DeclarationSort())
```

Comment:
Can we move `direction` function as apply in `DeclarationSort

Reply:
I think he's pulling it into a binder